backend/Dockerfile

```diff
@@ -8,7 +8,7 @@ RUN go mod vendor
 RUN GO111MODULE=on go build -o /bin/apiserver backend/src/apiserver/*.go
 
 # 2. Compile preloaded pipeline samples
-FROM python:3.5 as compiler
+FROM python:3.6 as compiler
```

Comment:
The latest python version is 3.9. Maybe we can use more advanced version like 3.7 or 3.8 if we think 3.9 is not quite stable.

Reply:
Thank you Xiyue! I will update it to 3.9 for latest stable python version.

Reply:
Based on the logs, Looks like python 3.8 doesn't work with ml-metadata 0.22.1. 
https://github.com/google/ml-metadata/issues/37
You can try to upgrade ml-medatadata to 0.24 version.

Reply:
Or maybe you can use python 3.6 or 3.7. I'm not sure why we lock tfx to 0.22 version(ml-metadata is locked via tfx).

Reply:
Thank you Xiyue! When you say tfx locks ml-metadata version, does that mean we need to update both tfx and ml-metadata versions if we want to update one of them? How do I check what ml-metadata version does tfx require in each release?

Reply:
I think tfx is the direct dependency we rely on and tfx will use ml-metadata https://github.com/kubeflow/pipelines/blob/f65391309650fdc967586529e79af178241b4c2c/backend/requirements.txt#L75
And we use https://github.com/kubeflow/pipelines/blob/master/backend/requirements.in to lock the dependency and we use https://github.com/kubeflow/pipelines/blob/master/backend/update_requirements.sh to generate requirements.txt.

Reply:
By the way, I suggest we also update python version here when we generate requirements.txt
https://github.com/kubeflow/pipelines/blob/master/backend/update_requirements.sh

Reply:
/hold 
Since we upgrade python to 3.6 in this version. Maybe we can change the python version we used in pipelines/backend/update_requirements.sh to keep it consistent.

Reply:
We will fix the python failure issue in https://github.com/kubeflow/pipelines/pull/5033. I think we can revert the change of python version in this PR. Feel free to unhold the pr when you revert the change of python version in this PR

Reply:
Thank you Xiyue for your explanation! It makes sense in terms of locking/pinning dependency by specifying tfx version. The PR #5033 for presubmit failure issue has been merged.